File: tests/llvm/gaudi/benchmark/cpp/overlay_blend_8.cc
#include <iostream>
#include <vector>
#include <chrono>

#include "utils.h"

using namespace std;
using namespace std::chrono;

vector<vector<float>> overlay_blend_8(vector<vector<float>> base, vector<vector<float>> active) {
    vector<vector<float>> out;
    int m = base.size();
    int n = base[0].size();
    for (int row = 0; row < m; row++) {
        vector<float> row_vec;
        for (int col = 0; col < n; col++) {
            float pixel;
			if (base[row][col] >= 128)
                pixel = 2 * base[row][col] + base[row][col] - 2 * base[row][col] * base[row][col] / 255 - 255;
			else
                pixel = 2 * base[row][col] * base[row][col] / 255;
			row_vec.push_back(pixel);
        }
        out.push_back(row_vec);
    }
    return out;
}


int main() {
    mat_timer(&overlay_blend_8);
    return 0;
}

File: tenspiler/benchmark/python/overlay_blend_8.py
from typing import List


def overlay_blend_8(base: List[List[int]], active: List[List[int]]) -> List[List[int]]:
    output: List[List[int]] = []
    m = len(base)
    n = len(base[0])
    for i in range(m):
        curr_row: List[int] = []
        for j in range(n):
            if base[i][j] >= 128:
                pixel = (
                    2 * base[i][i]
                    + base[i][j]
                    - 2 * base[i][j] * base[i][j] / 255
                    - 255
                )
            else:
                pixel = 2 * base[i][j] * base[i][j] / 255
            curr_row.append(pixel)
        output.append(curr_row)
    return output
